src/main/java/org/jboss/forge/arquillian/command/cube/CubeSetupCommand.java

```diff
@@ -0,0 +1,336 @@
+
+        parameters.put("definitionFormat", "CUBE");
+        parameters.put(cubeSetupFacet.getCubeConfiguration().getKeyForFileLocation(), yamlSnippet);
+    }
+
+    private Map<String, Object> getConfigParametersForDocker(UIExecutionContext context) {
+        Map<String, Object> params = new LinkedHashMap<>();
+        Map<String, String> build = new LinkedHashMap<>();
+
+        final String dirPath = getDockerDirectoryPath(context);
+
+        String imageParams = "dockerfileLocation: " + dirPath + System.lineSeparator() +
+            "noCache: true" + System.lineSeparator() +
+            "remove: true";
+
+        if (dockerFileName.hasValue()) {
+            imageParams += System.lineSeparator() + "dockerfileName: " + dockerFileName.getValue();
+        }
+
+        build.put("buildImage", imageParams);
+        params.put("containerName", build);
+        return params;
+    }
+
+    private String getDockerDirectoryPath(UIExecutionContext context) {
+        String dirPath = filePath.getValue();
+        String fileName = dockerFileName.getValue();
+        final int dirPathLength = dirPath.length();
+        if (dirPath.endsWith("Dockerfile")) {
+            dirPath = dirPath.substring(0, dirPathLength - "Dockerfile".length());
+        } else if (dockerFileName.hasValue() && dirPath.endsWith(fileName)) {
+            dirPath = dirPath.substring(0, dirPathLength - fileName.length());
+        } else if (dirPath.endsWith(File.separator)) {
+            dirPath = dirPath.substring(0, dirPathLength - 1);
+        }
+        final String fullyQualifiedName = getSelectedProject(context).getRoot().getFullyQualifiedName();
+        final File file = new File(fullyQualifiedName + File.separator +  dirPath);
+
+        if (file.isFile()) {
+                dirPath = dirPath.substring(0, dirPathLength - file.getName().length() -1);
+        }
+        return dirPath;
+    }
+
+    private String checkResourcesExists(UIExecutionContext context) throws IOException {
+        if (filePath.hasValue()) {
+            String filePath = this.filePath.getValue();
+            if (!Strings.isURL(filePath)) {
+
+                if (dockerFileName.hasValue()) {
+                    String dockerfileName = dockerFileName.getValue();
+                    if (!filePath.contains(dockerfileName)) {
+                        if (!filePath.endsWith(File.separator)) {
+                            filePath += File.separator;
+                        }
+                        filePath += dockerfileName;
+                    }
+                }
+
+                if (!fileExistsAndisNotDir(context, filePath)) {
+                    return "Could not find provided filePath: " + filePath + " or it is directory which is not allowed.";
+                }
+            }
+        }
+        return null;
+    }
+
+    private boolean fileExistsAndisNotDir(UIExecutionContext context, String fileName) {
```

Comment:
Can we move the whole file/resource logic to its own component and also unit test it?